flocker/node/agents/test/test_blockdevice.py

```diff
@@ -275,31 +277,187 @@ class BlockDeviceDeployerLocalStateTests(SynchronousTestCase):
     Tests for ``BlockDeviceDeployerLocalState``.
     """
     def setUp(self):
-        self.node_state = EMPTY_NODE_STATE
-        self.nonmanifest_datasets = NonManifestDatasets(datasets={})
-        self.volumes = []
-        self.local_state = BlockDeviceDeployerLocalState(
-            node_state=self.node_state,
-            nonmanifest_datasets=self.nonmanifest_datasets,
-            volumes=self.volumes
-        )
+        self.node_uuid = uuid4()
+        self.hostname = u"192.0.2.1"
 
     def test_provides_ilocalstate(self):
         """
         Verify that ``BlockDeviceDeployerLocalState`` instances provide the
         ILocalState interface.
         """
+        local_state = BlockDeviceDeployerLocalState(
+            node_uuid=self.node_uuid,
+            hostname=self.hostname,
+            datasets={},
+        )
         self.assertTrue(
-            verifyObject(ILocalState, self.local_state)
+            verifyObject(ILocalState, local_state)
         )
 
-    def test_shared_changes_are_shared(self):
+    def test_shared_changes(self):
         """
-        Verify that the attributes of ``BlockDeviceDeployerLocalState`` that
-        are supposed to be shared are returned from shared_state_changes().
+        ``shared_state_changes`` returns a ``NodeState`` with
+        the ``node_uuid`` and ``hostname`` from the
+        BlockDeviceDeployerLocalState`` and a
+        ``NonManifestDatasets``.
         """
-        self.assertEqual((self.node_state, self.nonmanifest_datasets),
-                         self.local_state.shared_state_changes())
+        local_state = BlockDeviceDeployerLocalState(
+            node_uuid=self.node_uuid,
+            hostname=self.hostname,
+            datasets={},
+        )
+        expected_changes = (
+            NodeState(
+                uuid=self.node_uuid,
+                hostname=self.hostname,
+                manifestations={},
+                paths={},
+                devices={},
+                applications=None
+            ),
+            NonManifestDatasets(
+                datasets={},
+            )
+        )
+        self.assertEqual(
+            local_state.shared_state_changes(),
+            expected_changes,
+        )
+
+    def test_non_manifest_dataset(self):
+        """
+        When there is a a dataset in the ``NON_MANIFEST`` state,
+        it is reported as a non-manifest dataset.
+        """
+        dataset_id = uuid4()
+        local_state = BlockDeviceDeployerLocalState(
+            node_uuid=self.node_uuid,
+            hostname=self.hostname,
+            datasets={
+                dataset_id: DiscoveredDataset(
+                    state=DatasetStates.NON_MANIFEST,
+                    dataset_id=dataset_id,
+                    blockdevice_id=ARBITRARY_BLOCKDEVICE_ID,
+                    maximum_size=LOOPBACK_MINIMUM_ALLOCATABLE_SIZE,
+                ),
+            },
+        )
+        expected_changes = (
+            NodeState(
```

Comment:
:grey_question: Consider putting this empty state in a member variable, and explicitly modifying it in future tests to what you are expecting. Concern about maintenance burden of adding a new parameter to NodeState, this would add 4 more separate places that we would have to run around to and add the parameter, instead of just 1.

I recognize that's probably just a matter of personal preference though so I'll leave it up to you.